### PR TITLE
py: reduce array slice assignment code size

### PR DESCRIPTION
**Pull request by dxxb**
Unix port on x86 compiles with or without MICROPY_PY_BUILTINS_MEMORYVIEW, executable sizes:

```
master
LINK micropython
   text    data     bss     dec     hex filename
 316973     756    1208  318937   4ddd9 micropython

dxb/slice_assign_diet
LINK micropython
   text    data     bss     dec     hex filename
 316893     756    1208  318857   4dd89 micropython
```

### CODE REVIEW

File: py/objarray.c [hunk 1]
```diff
@@ -402,37 +402,30 @@ STATIC mp_obj_t array_subscr(mp_obj_t self_in, mp_obj_t index_in, mp_obj_t value
...
-                    if (o->base.type == &mp_type_memoryview) {
+                uint8_t* dst_ptr = o->items;
+                #if MICROPY_PY_BUILTINS_MEMORYVIEW
+                if (o->base.type == &mp_type_memoryview) {
+                    if (len_adj != 0) {
                         goto compat_error;
                     }
-                    #endif
+                    dst_ptr += o->free * item_sz;
+				}
```
Comment (dpgeorge) [inline py/objarray.c]: No physical tabs!

File: py/objarray.c [hunk 2]
```diff
@@ -402,37 +402,30 @@ STATIC mp_obj_t array_subscr(mp_obj_t self_in, mp_obj_t index_in, mp_obj_t value
 
                 // TODO: check src/dst compat
                 mp_int_t len_adj = src_len - (slice.stop - slice.start);
-                if (len_adj > 0) {
-                    #if MICROPY_PY_BUILTINS_MEMORYVIEW
-                    if (o->base.type == &mp_type_memoryview) {
+                uint8_t* dst_ptr = o->items;
```
Comment (dpgeorge) [inline py/objarray.c]: I'd call it `dest_items` to be compatible with `src_items`.

**Comment by dpgeorge** [comment]
It's also good to check size change with Thumb2 arch (using eg stmhal), and in this case it saves 52 bytes, which is very nice!

File: py/objarray.c (hunk 1 above)
Comment (dxxb) [inline py/objarray.c]: Fixed

File: py/objarray.c (hunk 2 above)
Comment (dxxb) [inline py/objarray.c]: I made this change too

**Comment by dpgeorge** [comment]
Thanks!  Next time we can try and do it all in one PR :)

**Pull request merged**